cpp/libs/ossrf_nmos_api/lib/src/context/resource_map.cpp

```diff
@@ -68,3 +68,39 @@ expected<nmos_resource_ptr> resource_map_t::find_resource(const std::string& res
     BST_FAIL("didn't find any resource with ID {}", resource_id);
     return {};
 }
+
+std::vector<std::string> resource_map_t::get_sender_ids() const
+{
+    std::vector<std::string> ids;
+    lock_t lock(mutex_);
+    for(const auto& [id, resources] : map_)
+    {
+        for(const auto& resource : resources)
+        {
+            if(resource->get_resource_type() == nmos::types::sender)
+            {
+                ids.push_back(id);
+            }
+        }
+    }
+
+    return ids;
+}
+
+std::vector<std::string> resource_map_t::get_receiver_ids() const
+{
+    std::vector<std::string> ids;
+    lock_t lock(mutex_);
+    for(const auto& [id, resources] : map_)
+    {
+        for(const auto& resource : resources)
+        {
+            if(resource->get_resource_type() == nmos::types::receiver)
+            {
+                ids.push_back(id);
+            }
+        }
+    }
+
+    return ids;
+}
```

Comment:
Missing line here. Doesn't clang-format get this?

Reply:
Clang-format doesn't get this, but the line is now added